### PR TITLE
Bump velero library and deployment to 1.14 + go update and kubevirtci

### PR DESCRIPTION
**Pull request by ShellyKa13**
**Release note**:

```release-note
Update velero to 1.14 and go to 1.22.3
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,17 +1,19 @@
 module kubevirt.io/kubevirt-velero-plugin
 
-go 1.18
+go 1.21
```
Comment (akalenyu) [inline go.mod:3]: maybe just go to 1.22 already? the builder is anyway on this version

**Comment by akalenyu** [comment]
So I don't fully understand it yet but somehow the .gitignore directive for velero* introduced in https://github.com/kubevirt/kubevirt-velero-plugin/pull/208/commits/71c67b4ea7918955eee34ee8e9b2dd92784909d1
is dropping some critical additions in the velero vendored dir
```
$ git status  
On branch bump-velero
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   .gitignore

Untracked files:
  (use "git add <file>..." to include in what will be committed)
        vendor/github.com/vmware-tanzu/velero/pkg/apis/velero/shared/
        vendor/github.com/vmware-tanzu/velero/pkg/cmd/util/flag/orlabelselector.go
        vendor/github.com/vmware-tanzu/velero/pkg/features/verify.go
        vendor/github.com/vmware-tanzu/velero/pkg/plugin/framework/common/plugin_config.go

no changes added to commit (use "git add" and/or "git commit -a")
```

**Comment by alromeros** [comment]
Looks good, nice commit separation
/lgtm

File: hack/velero/deploy-velero.sh [hunk 2]
```diff
@@ -58,6 +58,8 @@ if [[ ! $(_kubectl get deployments -n velero | grep velero) ]]; then
     --bucket velero \
     --secret-file ${velero_resources_dir}/credentials-velero \
     --use-volume-snapshots=true \
+    --velero-pod-mem-request 512Mi \
+    --velero-pod-mem-limit 1Gi \
```
Comment (akalenyu) [inline hack/velero/deploy-velero.sh:62]: This is a high amount, are the OOMs maybe concerning to us/velero?
Maybe related to https://issues.redhat.com/browse/OCPBUGS-35436?

**Pull request merged**